src/test/java/school/hei/haapi/integration/StudentIT.java

```diff
@@ -215,21 +213,55 @@ void manager_write_update_ok() throws ApiException {
     assertTrue(updated.contains(toUpdate1));
   }
 
+  @Test
+  void manager_write_update_rollback_on_event_error() throws ApiException {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    UsersApi api = new UsersApi(manager1Client);
+    Student toCreate = aCreatableStudent();
+    reset(eventBridgeClientMock);
+    when(eventBridgeClientMock.putEvents((PutEventsRequest) any()))
+        .thenThrow(RuntimeException.class);
+
+    assertThrowsApiException(
+        "{\"type\":\"500 INTERNAL_SERVER_ERROR\",\"message\":null}",
+        () -> api.createOrUpdateStudents(List.of(toCreate)));
+
+    List<Student> actual = api.getStudents(1, 100);
+    assertFalse(actual.stream().anyMatch(s -> Objects.equals(toCreate.getEmail(), s.getEmail())));
+  }
+
+  @Test
+  void manager_write_update_more_than_10_students_ko() throws ApiException {
+    ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
+    UsersApi api = new UsersApi(manager1Client);
+    Student studentToCreate = aCreatableStudent();
+    List<Student> listToCreate = aStudentList(11);
+    listToCreate.add(studentToCreate);
+
+    assertThrowsApiException(
+        "{\"type\":\"400 BAD_REQUEST\",\"message\":\"students to create or update must be ≤ 10\"}",
+        () -> api.createOrUpdateStudents(listToCreate));
+
+    List<Student> actual = api.getStudents(1, 100);
+    assertFalse(
+        actual.stream().anyMatch(s -> Objects.equals(studentToCreate.getEmail(), s.getEmail())));
+  }
+
   @Test
   void manager_write_update_triggers_userUpserted() throws ApiException {
     ApiClient manager1Client = anApiClient(MANAGER1_TOKEN);
     UsersApi api = new UsersApi(manager1Client);
     reset(eventBridgeClientMock);
-    when(eventBridgeClientMock.putEvents((PutEventsRequest) any())).thenReturn(
-        PutEventsResponse.builder()
-            .entries(
-                PutEventsResultEntry.builder().eventId("eventId1").build(),
-                PutEventsResultEntry.builder().eventId("eventId2").build())
-            .build());
-
-    List<Student> created = api.createOrUpdateStudents(List.of(
-        aCreatableStudent(),
-        aCreatableStudent()));
+    when(eventBridgeClientMock.putEvents((PutEventsRequest) any()))
+        .thenReturn(
```

Comment:
Prevent yourself from doing such big changes if does not bring anything to the codebase.